CMakeLists.txt

```diff
@@ -620,12 +632,96 @@ foreach(tgt ${BINARY_TARGETS})
+   set(_LIBARCHIVE_CMAKE_ARGS "")
+ else()
+   set(_LIBARCHIVE_CMAKE_ARGS -DZSTD_INCLUDE_DIR=${PROJECT_SOURCE_DIR}/zstd/lib -DCMAKE_LIBRARY_PATH=${PROJECT_BINARY_DIR}/zstd/build/cmake/lib/)
+ endif()
+
+
+ExternalProject_Add(${_LIBARCHIVE_PRJ}
+  PREFIX "${PROJECT_BINARY_DIR}/libarchive"
+  GIT_REPOSITORY "https://github.com/libarchive/libarchive.git"
+  GIT_TAG "v3.5.1"
+  CMAKE_ARGS ${_LIBARCHIVE_CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}/libarchive -DCMAKE_BUILD_TYPE=Release -DENABLE_ICONV:BOOL=OFF -DENABLE_LIBXML2:BOOL=OFF
+  BUILD_BYPRODUCTS ${__LIBARCHIVE}
+  TMP_DIR       "${PROJECT_BINARY_DIR}/libarchive/tmp"
+  STAMP_DIR     "${PROJECT_BINARY_DIR}/libarchive/stamp"
+  SOURCE_DIR    "${PROJECT_BINARY_DIR}/libarchive/src"
+  BINARY_DIR    "${PROJECT_BINARY_DIR}/libarchive/build"
+  INSTALL_DIR   ""
+)
+
+  if(NOT (ZSTD_FOUND AND PREFER_SYSTEM_ZSTD))
+    add_dependencies(${_LIBARCHIVE_PRJ} libzstd_static)
+  endif()
 
+add_library(_LIBARCHIVE STATIC IMPORTED)
+set_target_properties(_LIBARCHIVE PROPERTIES IMPORTED_LOCATION ${__LIBARCHIVE})
+add_dependencies(_LIBARCHIVE ${_LIBARCHIVE_PRJ})
+
+# ...................................................................
+# Each library name given to the NAMES option is first considered
+# as a library file name and then considered with platform-specific 
+# prefixes (e.g. lib) and suffixes (e.g. .so). 
+# ...................................................................
+
+  find_library(_LIBGLOG "libglog.a" REQUIRED)
+  find_library(_LIBFMT "libfmt.a" REQUIRED)
+  find_library(_LIBDC "libdouble-conversion.a" REQUIRED)
+  find_library(_LIBGFLAGS "libgflags.a" REQUIRED)
+  find_library(_LIBEVENT "libevent.a" REQUIRED)
+  find_library(_LIBACL "libacl.a" REQUIRED) 
+  find_library(_LIBXML2 "libxml2.a" REQUIRED) 
+  find_library(_LIBCRYPTO "libcrypto.a" REQUIRED) 
+  find_library(_LIBZ "libz.a" REQUIRED) 
+  find_library(_LIBPTHREAD "libpthread.a" REQUIRED) 
+  find_library(_LIBDL "libdl.a" REQUIRED)   
+  find_library(_LIBC "libc.a" REQUIRED)  
+  find_library(_LIBM "libm.a" REQUIRED)
+  find_library(_LIBRT "librt.a" REQUIRED)
+  find_library(_LIBSSL "libssl.a" REQUIRED)
+  find_library(_LIBUNWIND "libunwind.a" REQUIRED)
+
+  foreach(tgt ${BINARY_TARGETS})
+
+# ...................................................................
+#  -static-libgcc above and gcc_eh below is all together 
+#   an ugly trick to enforce static linking
+# ...................................................................
+    target_link_libraries(${tgt}
+      ${_LIBFMT}          ${_LIBDC}           ${_LIBGLOG}         ${_LIBGFLAGS}                
+      _LIBARCHIVE         ${_LIBEVENT}        ${_LIBACL}          
+      ${_LIBSSL}          ${_LIBCRYPTO}       ${_LIBPTHREAD}      ${_LIBDL}
+      ${_LIBZ}            ${_LIBC}            ${_LIBM}            ${_LIBRT}
+      gcc_eh              ${_LIBUNWIND}             
+    )
+  endforeach()
+endif(STATIC_BUILD_DO_NOT_USE)
+# icui18n icuuc icudata
```

Comment:
What's this comment about?

Reply:
It used to be a list of libraries that were required by libarchieve under certain circumstances.
Just forgot to delete it.

Reply:
Removed